### DISCUSSION TITLE
Troubleshooting issue with automatic renewals (LetsEncrypt)

### DISCUSSION THREAD

**Discussion by tigreye007** [shibayan/appservice-acmebot#331] [Q&A] [answered]
According to the documentation, renewals should automatically occur 30 days prior to expiration, however we have yet to see this occur on the sites we've set up using this function app.  We have a few certificates expiring soon, and we're hoping to figure out why they aren't renewing.  I'm having trouble finding logs, or seeing any sort of scheduling for the renewal.  Is there any way to see which certificates have been generated by the app, and when they are scheduled to renew?  If the renewal function(s) is/aren't being run, is there a proper way to trigger the process and/or troubleshoot to figure out why?

Thanks!

**Comment by shibayan** [comment, marked as answer]
Auto-renewals are processed on Mondays and Fridays, so there will be a slight lag even if it is 30 days before the expiration date. If you want to perform the renewal process manually, you can do so by running the `RenewCertificates_Timer` function from Azure Portal.

It is recommended to set up a webhook to know that an error has occurred (see README.md)

**Reply by tigreye007** [reply]
Thanks for the note.  I was able to figure out that in both of these cases, the sites went through an upgrade of the base platform, however I'm not sure how the "configured" file and custom web.config disappeared.  I guess we'll need to keep an eye on upgrades going forward.  Once I re-uploaded the appropriate web.config and configured file and ran the RenewCertificates_Timer function, the certs were successfully renewed.

As for webhooks, do you have any documentation on how to set those up?  The readme currently only mentions webhook support, but nothing more.

**Reply by shibayan** [reply]
Acmebot will not delete the site/.well-known/configured file, so be careful not to delete it by mistake.

The README for Acmebot here did not include instructions on how to set up a webhook, so I fixed it.
https://github.com/shibayan/appservice-acmebot#2-add-application-settings
